bdk_core/src/chain_graph.rs

```diff
@@ -19,8 +19,8 @@ use core::fmt::Debug;
 /// from the graph.
 #[derive(Clone, Debug, PartialEq)]
 pub struct ChainGraph<I = TxHeight> {
-    pub chain: SparseChain<I>,
-    pub graph: TxGraph,
+    chain: SparseChain<I>,
```

Comment:
chain has a checkpoint limit that you need to be able to set. Either make setting it possible on the chain graph or make an issue about discussing this if you think it needs more thought.

Given all our changes are checkpoint limits still needed?

Reply:
I'm not sure, I'll expose the method to set checkpoint limits.